Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt

```diff
@@ -85,20 +85,34 @@ private fun Collection<CwaCovidCertificate>.rule2FindRecentRaCertificate(
  */
 private fun Collection<CwaCovidCertificate>.rule3FindRecentLastShot(
     nowUtc: Instant
-): CwaCovidCertificate? = this
-    .filterIsInstance<VaccinationCertificate>()
-    .filter {
-        with(it.rawCertificate.vaccination) { doseNumber == totalSeriesOfDoses }
+): CwaCovidCertificate? {
+    val twoDoseVaccines = listOf("EU/1/20/1528", "EU/1/21/1529", "EU/1/20/1507")
```

Comment:
Can we add a list of one dose vaccine (with Janssen as its only member for now)  and check against this list to make it more robust?

Reply:
...instead of checking if not in list with two doses

Reply:
Yes, sure.

Reply:
I suggest that you put these constants in the `VaccaintionCertificate` itself and/or add logic around these constants in it or an extension , whatever you think is better. These are the same as in
https://github.com/corona-warn-app/cwa-app-android/blob/49fa93bbd2896fd94d6b5335832ac19f7e9de72e/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPerson.kt#L122-L124

to have them in a common place